src/api/transportTypes.js

```diff
@@ -31,25 +31,47 @@ export type ApiResponseSuccess = {|
   msg: '',
 |};
 
-/** List of error codes at https://github.com/zulip/zulip/blob/master/zerver/lib/exceptions.py */
-export type ApiErrorCode =
-  | 'BAD_REQUEST'
-  | 'REQUEST_VARIABLE_MISSING'
-  | 'REQUEST_VARIABLE_INVALID'
-  | 'BAD_IMAGE'
-  | 'REALM_UPLOAD_QUOTA'
-  | 'BAD_NARROW'
-  | 'MISSING_HTTP_EVENT_HEADER'
-  | 'STREAM_DOES_NOT_EXIST'
-  | 'UNAUTHORIZED_PRINCIPAL'
-  | 'UNEXPECTED_WEBHOOK_EVENT_TYPE'
-  | 'BAD_EVENT_QUEUE_ID'
-  | 'CSRF_FAILED'
-  | 'INVITATION_FAILED'
-  | 'INVALID_ZULIP_SERVER';
+/**
+ * A list of current error codes can be found at:
+ *   https://github.com/zulip/zulip/blob/master/zerver/lib/exceptions.py
+ *
+ * Unfortunately, the `code` entry is a relatively late addition to the Zulip
+ * API: it hasn't always been present in responses, and even on current versions
+ * of the server, may still not be present in some. Since we also can't reliably
```

Comment:
In fact the situation is just like this except a bit more so: most errors don't have a `code`, and it's not clear that most ever will. From a grep in the server repo, first for things with codes and then without:
```
$ rg 'class.*\(.*JsonableError' | wc -l
20

$ rg 'JsonableError\(' | wc -l
212
```

I think that's probably even the right strategy: if we did try to make a policy of "all errors have codes", realistically it would lead to a lot of error codes with ill-thought-out semantics. Instead we add codes lazily, whenever there's a known use case, and so we do OK on actually thinking through the API choices of the code's name, the boundaries of what it covers, etc.